### ISSUE TITLE
Support taskENTER_CRITICAL?

### ISSUE THREAD

**Issue by srobertjames** [jonenz/FreeRTOS-Cpp#10]
This C++ wrapper looks excellent  - a simple wrapper that does exactly that without ruining FreeRTOS or doing more than I ask for.

Do you support `taskENTER_CRITICAL` https://www.freertos.org/taskENTER_CRITICAL_taskEXIT_CRITICAL.html ? I couldn't find any such methods.

**Comment by jonenz** [comment]
Yes. It's part of the kernel namespace.

[ FreeRTOS::Kernel::enterCritical()](https://jonenz.github.io/FreeRTOS-Cpp/namespaceFreeRTOS_1_1Kernel.html#a71df9b3921b7d8d1c25995a750f3609d)

**Comment by srobertjames** [comment]
Thank you.  For my knowledge, why is it there and not part of Task? What is the division between Kernel and Task? (I had expected things that begin with `task` prefix to be in Task class)

**Comment by jonenz** [comment]
The idea was to put things in the Task class operate on a specific instance of a task or take `TaskHandle_t` as an argument to the FreeRTOS function. Things in the Kernel namespace are more generic operations that apply to the entire kernel.

`taskENTER_CRITICAL()` is really masking interrupts from task context but doesn't have anything to do with a specific instance of a task other than ensuring the current context will continue to execute uninterrupted (up until `configMAX_SYSCALL_INTERRUPT_PRIORITY`).

`taskENTER_CRITICAL_FROM_ISR()` is masking interrupts from ISR context and has nothing to do with any tasks.

**Comment by jonenz** [comment]
Because `taskENTER_CRITICAL()` is intended to be called from task context, it may make sense to move it to a protected function of the task class by the same logic as `delay()` (see https://github.com/jonenz/FreeRTOS-Cpp/discussions/2), but it wouldn't make sense to do that for `taskENTER_CRITICAL_FROM_ISR()` which might be a bit confusing...

**Comment by milhead2** [comment]
Hi John,

Do you support the ENTER/EXIT critical with a scope_lock sort of class?

Mil

________________________________

**Comment by jonenz** [comment]
> Do you support the ENTER/EXIT critical with a scope_lock sort of class?

No, but I would be open to adding some of these implementations in another folder.
